iceoryx_posh/include/iceoryx_posh/popo/user_trigger.hpp

```diff
@@ -46,35 +46,23 @@ class UserTrigger
     /// @brief Resets the UserTrigger state to not triggered
     void resetTrigger() noexcept;
 
-    template <uint64_t>
-    friend class WaitSet;
+    /// @brief Only usable by the WaitSet, not for public use. Invalidates the internal triggerHandle.
+    /// @param[in] EventAccessor class constructable only by types which should have access to this method
+    /// @param[in] uniqueTriggerId the id of the corresponding trigger
+    void invalidateTrigger(const EventAccessor, const uint64_t uniqueTriggerId) noexcept;
 
-  private:
-    void invalidateTrigger(const uint64_t uniqueTriggerId) noexcept;
-
-    /// @brief enables the trigger event
-    /// @param[in] waitset reference to the waitset to which the UserTrigger should be attached
-    /// @param[in] eventId optional parameter, the id of the corresponding event
-    /// @param[in] callback optional parameter, the callback of the event
-    /// @return if the event could not be attached to the given waitset the expected contains the error, otherwise
-    /// the expected signals success
-    template <uint64_t WaitSetCapacity>
-    cxx::expected<WaitSetError> enableEvent(WaitSet<WaitSetCapacity>& waitset,
-                                            const uint64_t eventId = EventInfo::INVALID_ID,
-                                            const EventInfo::Callback<UserTrigger> callback = nullptr) noexcept;
+    /// @brief Only usable by the WaitSet, not for public use. Returns method pointer to UserTrigger::hasTriggered
+    /// @param[in] EventAccessor class constructable only by types which should have access to this method
+    WaitSetHasTriggeredCallback getHasTriggeredCallbackForEvent(const EventAccessor) const noexcept;
```

Comment:
I named it explicilty getHasTriggeredCallbackForEvent. Here the reason:
A WaitSet has to find out which trigger has triggered it. For this task it requires a callback to ask the trigger were you triggered or `trigger.hasTriggered()` in the code. And the corresponding callback for `hasTriggered` is the `hasTriggeredCallback`. Since a class can have multiple `hasTriggeredCallbacks` since it can have multiple events I named it `getHasTriggeredCallbackForEvent`. 

But I admit for a single event class it does not make sense and I will rename it only in the case of the user trigger into `getHasTriggeredCallback`.